joomla/joomla-cms
# [4.2] Resolve application variable when it doesn't exist in base plugin

Pull Request for Issue discovered in https://github.com/joomla/joomla-cms/pull/38060#issuecomment-1167021466.

### Summary of Changes
Makes a proxy for the app variable in the `CMSPlugin` class. It helps in the transition to the service providers where the application should be injected and not magically resolved by class reflection and static Factory access.

Wondering if we should also handle the `db` variable.

Pink here @nikosdion for a review.

### Testing Instructions
Create a demo sleep task. There is the app lookup done. Or revert the patch from #38132 and run the testing instructions, they should still work.

### Actual result BEFORE applying this Pull Request
A deprecated warning is not logged, but the task works as before.

### Expected result AFTER applying this Pull Request
A deprecated warning is logged, but the task works as before.
